src/astro/databases/base.py

```diff
@@ -0,0 +1,259 @@
+    Database module name. In that case, we should update the dictionary
+    `CUSTOM_CONN_TYPE_TO_MODULE_PATH` available at `astro/databases/__init__.py`.
+    """
+
+    _create_schema_statement: str = "CREATE SCHEMA IF NOT EXISTS {}"
+    _drop_table_statement: str = "DROP TABLE IF EXISTS {}"
+    _create_table_statement: str = "CREATE TABLE IF NOT EXISTS {} AS {}"
+
+    def __init__(self, conn_id: str):
+        self.conn_id = conn_id
+
+    @property
+    def hook(self) -> BaseHook:
+        """
+        Return an instance of the database-specific Airflow hook.
+        """
+        raise NotImplementedError
+
+    @property
+    def connection(self) -> sqlalchemy.engine.base.Connection:
+        """
+        Return a Sqlalchemy connection object for the given database.
+        """
+        return self.sqlalchemy_engine.connect()
+
+    @property
+    def sqlalchemy_engine(self) -> sqlalchemy.engine.base.Engine:
+        """
+        Return Sqlalchemy engine.
+        """
+        return self.hook.get_sqlalchemy_engine()
+
+    def run_sql(self, sql_statement: str, parameters: Optional[dict] = None):
+        """
+        Return the results to running a SQL statement.
+
+        Whenever possible, this method should be implemented using Airflow Hooks,
+        since this will simplify the integration with Async operators.
+
+        :param sql_statement: Contains SQL query to be run against database
+        :param parameters: Optional parameters to be used to render the query
+        """
+        return self.hook.run(sql_statement, parameters)
+
+    # ---------------------------------------------------------
+    # Table metadata
+    # ---------------------------------------------------------
+    def get_table_qualified_name(self, table: Table) -> str:
+        """
+        Return table qualified name. This is Database-specific.
+        For instance, in Sqlite this is the table name. In Snowflake, however, it is the database, schema and table
+
+        :param table: The table we want to retrieve the qualified name for.
+        """
+        # Initially this method belonged to the Table class.
+        # However, in order to have an agnostic table class implementation,
+        # we are keeping all methods which vary depending on the database within the Database class.
+        raise NotImplementedError
+
+    # ---------------------------------------------------------
+    # Table creation & deletion methods
+    # ---------------------------------------------------------
+    def create_table(self, table: Table) -> None:
+        """
+        Create a SQL table. For this method to work, the table instance must contain columns.
+
+        :param table: The table to be created.
+        """
+        metadata = table.sqlalchemy_metadata
+        sqlalchemy_table = sqlalchemy.Table(table.name, metadata, *table.columns)
+        metadata.create_all(self.sqlalchemy_engine, tables=[sqlalchemy_table])
```

Comment:
Might be worth it for this to return an `astro.sql.table.Table` object

Reply:
What do you think would be the benefit of this? The parts of the code which will use this method know `table` in advance, and pass it as an argument.